diff --git a/index.d.ts b/index.d.ts
index c0762387a68005990e4d2297ff2331e73c4e71ba..7fff5f90fe549b46a0c500ec0b343195f5988dd6 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -1 +1,4 @@
-import { Component, ReactNode } from "react";
+/* TODO: Remove this patch once upstream declarations use compatible React Context and Portal types.
+ * https://github.com/gtkx-org/gtkx/issues/752
+ */
+import { Component, Context, ReactNode, ReactPortal as PublicReactPortal } from "react";
@@ -649,18 +652 @@ declare namespace ReactReconciler {
-    interface ReactContext<T> {
-        $$typeof: symbol | number;
-        Consumer: ReactContext<T>;
-        Provider: ReactProviderType<T>;
-        _currentValue: T;
-        _currentValue2: T;
-        _threadCount: number;
-        // DEV only
-        _currentRenderer?: {
-            [key: string]: any;
-        } | null;
-        _currentRenderer2?: {
-            [key: string]: any;
-        } | null;
-        // This value may be added by application code
-        // to improve DEV tooling display names
-        displayName?: string;
-    }
+    interface ReactContext<T> extends Context<T> {}
@@ -668 +654 @@ declare namespace ReactReconciler {
-    interface ReactPortal {
+    interface ReactPortal extends PublicReactPortal {
@@ -952 +938 @@ declare namespace ReactReconciler {
-        ): ReactPortal;
+        ): PublicReactPortal;
